### ISSUE TITLE
Response is undefined when trying to copy a drive item.

### ISSUE THREAD

**Issue by nikithauc** [microsoftgraph/msgraph-sdk-serviceissues#34]
Thank you @victorreis for reporting this [issue](https://github.com/microsoftgraph/msgraph-sdk-javascript/issues/344#issuecomment-719645580). 

Steps to reproduce - 

- Try the [sample code](https://docs.microsoft.com/en-us/graph/api/driveitem-copy?view=graph-rest-1.0&tabs=javascript) to copy a drive item.
- Console log the response. 
- The response is empty.

**Comment by victorreis** [comment]
@nikithauc , something curious happens when I tried this code.

At the first tests it works fine, the copy was made and the `res.headers.location` was returning the URL correctly. Every time I was deleting the file and copying through this code. So I started to do my error handling to the case of copy conflict. Naturally, I had to do a lot of tests. At some point the location passed to bring me `undefined`. This happens even with `statusText === 'Accepted'` and the copy made. Without the URL I can't get the drive item ID. 

I think this is a problem with the API or maybe I was blocked by excessive tests.

Here is the code and the 
![image](https://user-images.githubusercontent.com/8744178/99257328-a09c4a00-27f5-11eb-98e7-5551829f0ac2.png)

**Comment by nikithauc** [comment]
@victorreis  Thank you for letting me know about this! 
I will check and get back to you.

**Comment by nikithauc** [comment]
@ificator Can you please help here with this issue?

**Comment by ificator** [comment]
It would definitely be unexpected for us to return a `202` response without a location header. Can you dump out all of the response headers when this occurs and add them to this issue? At the very least I'll need the `request-id` and `Date` headers to investigate.

**Comment by nikithauc** [comment]
@ificator  We get a location header with a 202 response. There are two issues  which are as follows -
1. The content-type is text/plain in the response returned from the Graph API and the body is empty.
Workloads shouldn't return a Text/plain content type if they are not going to return a body.
2. https://github.com/microsoftgraph/msgraph-sdk-serviceissues/issues/34#issuecomment-728027043

**Comment by ificator** [comment]
Point 2 suggests no location header doesn't it? Unless I'm misreading something.

But you're right, we should not set a content-type header if there's no body. It looks like the workload (SharePoint) is doing the right thing in that regard, and it's the Graph service itself that is adding the unexpected `Content-Type` header. We'll need that team to jump in and take a look.

**Comment by nikithauc** [comment]
@ificator  You are right about point 2. Sorry for my confusion. @victorreis  Please provide the information mentioned in https://github.com/microsoftgraph/msgraph-sdk-serviceissues/issues/34#issuecomment-730625567. That would helpful!

I will check with someone from the Graph Service team about the first problem.

**Comment by victorreis** [comment]
@nikithauc and @ificator, here it is:
![image](https://user-images.githubusercontent.com/8744178/99810282-44516700-2b22-11eb-88d7-4b10f4943132.png)

If I do the request in Postman it works fine. See:
![image](https://user-images.githubusercontent.com/8744178/99810760-f38e3e00-2b22-11eb-91a9-41ebac6b34ca.png)

**Comment by ificator** [comment]
Thanks @victorreis - any chance you could make the headers print out the values instead of `[Array]`? I'll need to see the actual values to investigate.

**Comment by victorreis** [comment]
> Thanks @victorreis - any chance you could make the headers print out the values instead of `[Array]`? I'll need to see the actual values to investigate.

@ificator , I don't know if I'm doing something wrong (please tell me if this is the case), but all the values listed with `[Array]` are undefined. 
![image](https://user-images.githubusercontent.com/8744178/99827212-20008500-2b38-11eb-850f-4e5f489af8aa.png)

**Comment by ificator** [comment]
A lot of those are probably inaccessible due to CORS (assuming you're running in a browser and not node.js), but location shouldn't fall into that category. I'm no expert in JavaScript / TypeScript and so if you give me a network trace (HAR, Fiddler, etc) I'd be able to make statements about whether you're seeing expected behaviour. A screenshot from Chrome / Edge / etc devtools that show the response headers for the request would also suffice.

**Comment by nikithauc** [comment]
@victorreis  
1. Can you give me the step to reproduce the problem you are facing? You mentioned here https://github.com/microsoftgraph/msgraph-sdk-serviceissues/issues/34#issuecomment-728027043
that you got the expected result at first but failed after multiple iterations. Can you please tell me what were the steps you were trying? 

2. Are you trying this operation on only one drive item? Can you reproduce this problem with multiple or other drive items?

**Comment by MIchaelMainer** [comment]
> At the first tests it works fine, the copy was made and the res.headers.location was returning the URL correctly. Every time I was deleting the file and copying through this code. So I started to do my error handling to the case of copy conflict. Naturally, I had to do a lot of tests. At some point the location passed to bring me undefined. This happens even with statusText === 'Accepted' and the copy made. Without the URL I can't get the drive item ID.

I'd like to add to Nikitha's question. In the case of the tests, are deleting and copying a file of the same name? Are you doing so each operation one after another? If you wait 1s, 5s, or 10s, do you see the same behavior?

**Comment by petrhollayms** [comment]
Thank you for reporting this issue. This appears to be an issue or limitation with the service APIs. Unfortunately, as the Microsoft Graph SDK team, we do not have ownership of the APIs that are causing you issues. We invite you to create a question about the service API to Microsoft Q&A and tagged with one of the [microsoft-graph-*] tags, that way it will get routed to the appropriate team for them to triage:  
 
https://aka.ms/msgraphsupport or directly https://aka.ms/askgraph 

For now, we will close the issue on our side but feel free to open it in the relevant repository if you think the issue is specific to SDK. 
Please let us know if this helps! 

Note: We will close this repository on April 19, 2024.
